### DISCUSSION TITLE
In case of Insert or Upsert, only insert or update (if document exists) those fields that are not None?

### DISCUSSION THREAD

**Discussion by jeanmonet** [BeanieODM/beanie#322] [Question]
Hi, I'm wondering if there's an option to specify that upon `save` or `upsert`, only those fields that are **not `None`** should be added / updated?  Or if there's a mechanism built into Beanie that could help with such use-case?

**Comment by roman-right** [comment]
I plan to add "exclude Nones" and "exclude defaults" options to the document config to be able to not store them in the database. For now it is not supported

**Comment by chximn** [comment]
If you happen to find this, I want to let you know that the feature has been added. The documentation now includes a part that explains how to do this using the `Settings` subclass in the document definition. You can check out the details at this link: [https://beanie-odm.dev/tutorial/defining-a-document/#keep-nulls](https://beanie-odm.dev/tutorial/defining-a-document/#keep-nulls)
